util/src/main/java/org/datacommons/util/TmcfCsvParser.java

```diff
@@ -167,18 +185,44 @@ public void process(CSVRecord dataRow) {
       }
     }
 
-    private Mcf.McfGraph.Values parseValues(Mcf.McfGraph.Values templateValues, CSVRecord dataRow) {
+    private Mcf.McfGraph.Values parseValues(
+        String templateEntity,
+        String currentProp,
+        Mcf.McfGraph.Values templateValues,
+        CSVRecord dataRow) {
       Mcf.McfGraph.Values.Builder instanceValues = Mcf.McfGraph.Values.newBuilder();
+
+      // Used for parseSchemaTerm() and splitAndStripWithQuoteEscape()
+      BiConsumer<String, String> errCb =
+          (counter, message) -> {
+            addLog(Debug.Log.Level.LEVEL_ERROR, counter, message);
+          };
+      BiConsumer<String, String> warnCb =
+          (counter, message) -> {
+            addLog(Debug.Log.Level.LEVEL_WARNING, counter, message);
+          };
+
       for (Mcf.McfGraph.TypedValue typedValue : templateValues.getTypedValuesList()) {
         if (typedValue.getType() == Mcf.ValueType.TABLE_ENTITY) {
-          assert !currentProp.equals(Vocabulary.DCID)
-              : "Unexpected value for dcid " + typedValue.getValue();
-          String referenceNode = toNodeName(typedValue.getValue());
+          if (currentProp.equals(Vocabulary.DCID)) {
+            addLog(
+                Debug.Log.Level.LEVEL_ERROR,
+                "CSV_TmcfEntityAsDcid",
+                "dcid property in TMCF entity " + templateEntity + " must not be an entity");
+            continue;
+          }
+          String referenceNode = toNodeName(typedValue.getValue(), errCb);
           Mcf.McfGraph.TypedValue.Builder newTypedValue = Mcf.McfGraph.TypedValue.newBuilder();
           if (referenceNode.startsWith(Vocabulary.DCID_PREFIX)) {
             String dcid = referenceNode.substring(Vocabulary.DCID_PREFIX.length());
             if (dcid.isEmpty()) {
-              incrementCounter("NumEmptyDCIDValues", logCtx);
+              addLog(
+                  Debug.Log.Level.LEVEL_WARNING,
+                  "CSV_EmptyDcidReferences",
+                  "Empty dcid reference in property "
```

Comment:
maybe mention the (required) missing dcid prefix?

Reply:
This is an edge case where value (reference) is missing... aka, there was `dcid:`

Reply:
oh did not get that from the message... maybe "entity dcid:{node} referenced in TMCF entity {entity} was not found"

Reply:
Clarified more. (This is the case where {node} is empty, rather than not found.)